### PR TITLE
Nominator: Nominate TX set hashes

### LINKED ISSUE #3211: Nominator: Nominate hash of the Tx set instead of set of TX hashes
as the load on the network increases we get bigger and bigger envelopes since we need to communicate more TX hashes.
Furthermore, more TXs in the network means more unique values nominated because of desync pools across different nodes.

So our network traffic increases worse than linearly, compared to the increase in TX count we are trying to externalize.

So instead of nominating Set!Hash for tx set, we should be nominating a single hash of the whole TX set and allow peers to download that TX set from us.

### PR DESCRIPTION
**Pull request by omerfirmak**
This will help scaling to higher TX counts easier. Envelopes wont
grow when nominated TX count increases.

Fixes #3211

### CODE REVIEW

**Comment by hewison-chris** [comment]
I triggered re run of failed CI

**Comment by hewison-chris** [comment]
Needs rebase I think.

File: source/agora/consensus/protocol/Data.d [hunk 1]
```diff
@@ -21,7 +21,7 @@ import agora.crypto.Hash;
 public struct ConsensusData
 {
     /// The transaction set that is being nominated / voted on
-    public Hash[] tx_set;
+    public Hash tx_set;
```
Comment (linked0) [inline source/agora/consensus/protocol/Data.d:24]: Can't we have `tx_set_hash` for that instead?

**Review by linked0 — COMMENTED** [review]
Very good attempt!

**Comment by omerfirmak** [comment]
All green and rebased

**Review by mkykadir — APPROVED** [review]

**Pull request merged**